force-app/main/default/classes/NextStepFuture.cls

```diff
@@ -29,48 +29,75 @@ public with sharing class NextStepFuture {
      */
     //[public | private ] [static] data_type method_name (input parameters) {
     //  The body of the method
-    //}
-
-    /*
-    * Question 2
-    * Update/Uncomment the template code to create a method
-    * Create a method that inserts a big deal opportunity and updates the user role name to 'CRO'
-    * Update the corresponding test class AsynchronousTest and test method nextStepFuture_testCreateDealAndUpdateRole to cover test this method
-    */
-    //Mix DML Error happens when you try to perform DML operations on setup and non-setup objects in a single transaction.
-    public static void createDealAndUpdateRole() {
-        Opportunity bigDeal = new Opportunity();
-        insertBigDeal(bigDeal);
-
-        UserRole[] roles = [SELECT Id, Name FROM UserRole WHERE Id = :UserInfo.getUserRoleId() OR Name = 'CRO' LIMIT 1];
-        updateUserRoleName(roles);
+    //
+    
+    
+    @future
+    public static void updateAccOpp(List<Id> accIds, String newNextStepField) {
+        // Query and update Accounts
+        try {
+            if (!accIds.isEmpty()) {
+                // Update Accounts
+                List<Account> accountsToUpdate = [SELECT Id, Next_Step__c FROM Account WHERE Id IN :accIds];
+                for (Account acc : accountsToUpdate) {
+                    acc.Next_Step__c = newNextStepField;
+                }
+                if (!accountsToUpdate.isEmpty()) {
+                    update accountsToUpdate;
+                }
+    
+                // Update Opportunities
+                List<Opportunity> opportunitiesToUpdate = [SELECT Id, NextStep FROM Opportunity WHERE AccountId IN :accIds];
```

Comment:
It's a little advanced but you can do a subquery for the opps within the query for the accounts.